cogs/calendar.py

```diff
@@ -0,0 +1,123 @@
+from icalevents.icalevents import events
+from discord.ext.commands import Cog
+from aiocron import crontab
+from aiohttp import ClientSession
+from json import loads, dumps
+from environs import Env
+
+
+class Calendar(Cog, name="Calendar"):
+    def __init__(self, bot) -> None:
+        """
+        Initializes the Calendar cog
+        :param bot: Bot that the cog shall attach to
+        """
+        env = Env()
+        env.read_env()
+        self.bot = bot
+        self.channel_id = env("CHANNEL_ID")
+        self.base_api_url = "https://discord.com/api/v8"
+        self.auth_headers = {
+            "Authorization": f"Bot {bot.token}",
+            "User-Agent": "DiscordBot Python/3.9 aiohttp/3.8.1",
+            "Content-Type": "application/json",
+        }
+        self.event_url = f"{self.base_api_url}/guilds/{bot.guild}/scheduled-events"
+
+    # crontab that runs daily
+    @crontab("0 0 * * *")
+    async def createEvent(self) -> None:
+        """
+        Creates events in the guild if they don't exist based on an iCal.
+        S
+        """
+        iCalEvents = events(self.bot.iCal_url)
+        serverEvents = await self.getGuildEvents()
+        for iEvent in iCalEvents:
```

Comment:
Maybe some filtering should be done here. No events from the past. How does it work with a series?

Reply:
While testing the function only loaded the upcoming 7 days. I did not found any information about this in the _icalevents_ documentation. I agree we should test this further and either adding a parameter or a comment to the function

Reply:
Ah okay. The comment was just from reading the code, not from running it. Can you create events further in the future than 7 days on discord?

Reply:
It is possible to supply _start_ and _end_ date. The function `events()` loads the whole calendar and safes it in _.cache_ as .txt if no range provided. The object that gets returned from the function however only holds the upcoming 7 days from observation, but is not stated in documentation.

https://icalevents.readthedocs.io/en/latest/

Reply:
Added parameters to function call.